### ISSUE TITLE
Upgrade SQLAlchemy version to >2.0.0

### ISSUE THREAD

**Issue by GGP1** [wazuh/wazuh#16679]
The current version of sqlalchemy (`1.3.11`) used in the framework module leaks cleartext passwords in logs and prints. This was fixed in version 2.0.0, we should upgrade the library to that version or a higher one.

Related discussion: https://github.com/sqlalchemy/sqlalchemy/discussions/8547
Fix: https://github.com/sqlalchemy/sqlalchemy/pull/8563

**Comment by fdalmaup** [comment]
# Issue Update

Despite the `sqlite3.NotSupportedError: deterministic=True requires SQLite 3.8.3 or higher` messages, the `SQLite` version from the Wazuh developed library is `3.36.0`. 

I tested installing `SQLAlchemy` version `2.0.0` in Wazuh `v4.3.10` and `v4.4.1` which have Python `3.9` in the embedded interpreter. After that, I executed the `mitredb.py` script and the only messages shown were warnings related to some methods being legacy methods. Therefore, the error may come from the `SQLite` linking when compiling the Python `3.10` interpreter.

I tried modifying the `SQLite` section in the `setup.py` script from `cpython` but the error still shows when executing the  `mitredb.py` script. I will keep investigating.

**Comment by fdalmaup** [comment]
# Issue Update

## Error root cause
The problem was that `/usr/include` contained a version in the `sqlite3.h` file lower than `3.8.3` but higher than the minimum allowed version by the CPython `setup.py` file (`MIN_SQLITE_VERSION_NUMBER`). So when CPython was compiled and found the system's version, which was greater than or equal to the minimum, it stayed with the system one and not with the one developed by Wazuh (currently version `3.36.0`). 

If we want the `/usr/include` directory not to be considered, we must modify the CPython `setup.py` code so that it only looks in `sqlite_inc_paths` (variable edited to contain the values of the Wazuh source directories). This will avoid any issues related to SQLite and the system's libraries.

## Installation in `CentOS 7 amd64`

After the implementation of the proposed solution, a successful installation of Wazuh was carried out with the new dependencies. Nevertheless, some warning messages were displayed due to the use of legacy methods:
```
/wazuh/tools/mitre/mitredb.py:27: MovedIn20Warning: The ``declarative_base()`` function is now available as sqlalchemy.orm.declarative_base(). (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
  Base = declarative_base()
/wazuh/tools/mitre/mitredb.py:701: LegacyAPIWarning: The Query.get() method is considered legacy as of the 1.x series of SQLAlchemy and becomes a legacy construct in 2.0. The method is now available as Session.get() (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
  technique = session.query(Technique).get(table[0])
/wazuh/tools/mitre/mitredb.py:697: LegacyAPIWarning: The Query.get() method is considered legacy as of the 1.x series of SQLAlchemy and becomes a legacy construct in 2.0. The method is now available as Session.get() (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
  software = session.query(Software).get(table[0])
/wazuh/tools/mitre/mitredb.py:689: LegacyAPIWarning: The Query.get() method is considered legacy as of the 1.x series of SQLAlchemy and becomes a legacy construct in 2.0. The method is now available as Session.get() (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
  groups = session.query(Group).get(table[0])
/wazuh/tools/mitre/mitredb.py:705: LegacyAPIWarning: The Query.get() method is considered legacy as of the 1.x series of SQLAlchemy and becomes a legacy construct in 2.0. The method is now available as Session.get() (deprecated since: 2.0) (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
  technique = session.query(Technique).get(table[0])
```
The next tasks are to investigate these messages and propose the necessary changes to the code.

**Comment by fdalmaup** [comment]
# Issue Update

Since SQLAlchemy 2.0, the standard approach in SQLAlchemy 2.0 is the use of [Session.execute()](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.execute) in conjunction with [select()](https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.select) to run ORM queries, instead of using [Session.query()](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.query). The [Query](https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html#sqlalchemy.orm.Query) API is not being removed from SQLAlchemy, as it is now implemented by using the new API internally it will remain as a legacy API, and both APIs can be used freely. So in case of being used in order sections of the code, it won't generate errors but warning messages.

The required changes in the `tools/mitre/mitredb.py` file were made in order to avoid the warning messages during the Wazuh installation. 
Next steps: 
- Review the code where the SQLAlchemy dependency is used to avoid possible warning messages.

**Comment by fdalmaup** [comment]
# Issue Update

Following the [mentioned migration guide](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html), the following files are subject to modifications to avoid warning messages:
- `framework/wazuh/rbac/orm.py`
- `wodles/azure/orm.py`

After the modifications, the related integration tests and/or unit tests will be executed to verify the correct functioning.

**Comment by fdalmaup** [comment]
# Issue Update

The `framework/wazuh/rbac/` tests give the following results:

**Comment by fdalmaup** [comment]
# Issue Update

Some changes have been applied to the code. The origin of the warnings seems to come from the use of the `Association Object Pattern` described for version `1.3` [here](https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#association-object) and updated for `2.0` [here](https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#combining-association-object-with-many-to-many-access-patterns).

Further changes may be required.

**Comment by fdalmaup** [comment]
# Issue update

The code was modified making use of the `overlaps` parameter in the involved relationships as suggested in the warning messages:
```
To silence this warning, add the parameter 'overlaps="roles_associations,users"' to the 'User.roles' relationship.
```

After running the `framework` tests, these are the warnings raised:
<details><summary>Remaining Warnings</summary>

    ======================================================================= warnings summary ========================================================================
    ../../venv/deps-3.10/lib/python3.10/site-packages/pytest_aiohttp/plugin.py:28
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/pytest_aiohttp/plugin.py:28: DeprecationWarning: The 'asyncio_mode' is 'legacy', switching to 'auto' for the sake of pytest-aiohttp backward compatibility. Please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
        config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
    
    framework/wazuh/core/utils.py:956
      /home/fdalmau/git/wazuh/framework/wazuh/core/utils.py:956: DeprecationWarning: invalid escape sequence '\g'
        data = re.sub(r'^&backslash;<(.*[^>])$', '&backslash;&lt;\g<1>', data)
    
    framework/wazuh/core/database.py:19
    framework/wazuh/core/database.py:19
    wazuh/core/tests/test_database.py::test_sql_compatibility
    wazuh/core/tests/test_database.py::test_sql_compatibility
      /home/fdalmau/git/wazuh/framework/wazuh/core/database.py:19: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
        if LooseVersion(sqlite3.sqlite_version) < LooseVersion('3.7.0.0'):
    
    framework/wazuh/core/InputValidator.py:18
      /home/fdalmau/git/wazuh/framework/wazuh/core/InputValidator.py:18: DeprecationWarning: invalid escape sequence '\w'
        """Abstract function to check a name matches a regex (\\w+ by default).
    
    framework/wazuh/core/cluster/dapi/tests/test_dapi.py:67
      /home/fdalmau/git/wazuh/framework/wazuh/core/cluster/dapi/tests/test_dapi.py:67: PytestCollectionWarning: cannot collect test class 'TestingLoggerParent' because it has a __init__ constructor (from: wazuh/core/cluster/dapi/tests/test_dapi.py)
        class TestingLoggerParent:
    
    framework/wazuh/core/cluster/dapi/tests/test_dapi.py:73
      /home/fdalmau/git/wazuh/framework/wazuh/core/cluster/dapi/tests/test_dapi.py:73: PytestCollectionWarning: cannot collect test class 'TestingLogger' because it has a __init__ constructor (from: wazuh/core/cluster/dapi/tests/test_dapi.py)
        class TestingLogger:
    
    scripts/tests/test_agent_groups.py::test_show_groups
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:42: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls([call(func=agent.get_agent_groups, kwargs={}),
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_show_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:69: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.get_agents, f_kwargs={'agent_list': agent_id}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_show_synced_agent
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:98: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.get_agents_sync_group, f_kwargs={'agent_list': [agent_id]}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_show_agents_with_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:125: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.get_agents_in_group, f_kwargs={'group_list': 'testing'}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_show_group_files
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:152: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.get_group_files, f_kwargs={'group_list': 'testing'}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_unset_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:181: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.remove_agent_from_groups,
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_remove_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:217: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.delete_groups, f_kwargs={'group_list': ['testing']}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_set_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:252: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.assign_agents_to_group,
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_agent_groups.py::test_create_group
      /home/fdalmau/git/wazuh/framework/scripts/tests/test_agent_groups.py:291: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        forward_mock.has_calls(call(func=agent.create_group, f_kwargs={'group_id': group_id}))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_cluster_control.py::test_main
      /usr/lib/python3.10/unittest/mock.py:333: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        self.__dict__[_the_name] = value
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    scripts/tests/test_rbac_control.py::test_restore_default_passwords[]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04aeb63400>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/core/cluster/tests/test_client.py::test_ac_client_echo_ok
      /usr/lib/python3.10/inspect.py:2224: RuntimeWarning: coroutine 'Handler.log_exceptions' was never awaited
        for i, (name, default) in enumerate(reversed(list(iter))):
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_client.py::test_ac_client_echo_ok
      /usr/lib/python3.10/inspect.py:2224: RuntimeWarning: coroutine 'Handler.send_request' was never awaited
        for i, (name, default) in enumerate(reversed(list(iter))):
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_common.py::test_response_init
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'test_run_in_pool.<locals>.LoopMock.run_in_executor' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_common.py::test_rst_done_callback
      wazuh/core/cluster/tests/test_common.py:261: PytestWarning: The test <Function test_rst_done_callback> is marked with '@pytest.mark.asyncio' but it is not an async function. Please remove asyncio marker. If the test is not marked explicitly, check for global markers applied via 'pytestmark'.
        @pytest.mark.asyncio
    
    wazuh/core/cluster/tests/test_common.py::test_rft_init
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ae9cf6d0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/core/cluster/tests/test_common.py::test_handler_update_chunks_wdb
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'Response.read' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_common.py::test_handler_wait_for_file_ko
    wazuh/core/cluster/tests/test_common.py::test_wazuh_common_error_receiving_file_ko
    wazuh/core/cluster/tests/test_master.py::test_master_handler_hello_ok
    wazuh/core/cluster/tests/test_master.py::test_master_handler_integrity_check[compare_result1]
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'Event.wait' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_common.py::test_handler_wait_for_file_ko
      /usr/lib/python3.10/json/encoder.py:257: RuntimeWarning: coroutine 'Event.wait' was never awaited
        return _iterencode(o, 0)
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_client.py::test_localclient_send_file
    wazuh/core/cluster/tests/test_worker.py::test_worker_handler_setup_receive_files_from_master
    wazuh/core/tests/test_configuration.py::test_get_active_configuration[000-agent-agent-sockets-ok {"agent": {"enabled": "yes"}}]
      /usr/lib/python3.10/unittest/mock.py:2181: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        self.name = name
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_server.py::test_LocalServerHandlerMaster_send_file_request
      /usr/lib/python3.10/unittest/mock.py:1412: RuntimeWarning: coroutine 'Handler.log_exceptions' was never awaited
        original = getattr(target, name, DEFAULT)
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_server.py::test_LocalServerHandlerMaster_send_file_request
      /usr/lib/python3.10/unittest/mock.py:1412: RuntimeWarning: coroutine 'test_LocalServerHandler_get_send_file_response.<locals>.func_mock' was never awaited
        original = getattr(target, name, DEFAULT)
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_server.py::test_LocalServerHandlerMaster_send_file_request
      /usr/lib/python3.10/unittest/mock.py:1412: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        original = getattr(target, name, DEFAULT)
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_server.py::test_LocalServerMaster_init
    wazuh/core/cluster/tests/test_master.py::test_rit_set_up_coro
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_local_server.py::test_LocalServerHandlerWorker_process_request
      /usr/lib/python3.10/unittest/mock.py:333: RuntimeWarning: coroutine 'Handler.log_exceptions' was never awaited
        self.__dict__[_the_name] = value
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_master.py::test_rit_set_up_coro
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'Handler.log_exceptions' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_master.py::test_master_handler_execute_ok
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/pytest_asyncio/plugin.py:375: RuntimeWarning: coroutine 'test_LocalServerHandlerMaster_send_file_request.<locals>.func_mock' was never awaited
        old_loop.close()
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_master.py::test_master_handler_get_nodes
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'Handler.forward_dapi_response' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_worker.py::test_rit_done_callback
      /usr/lib/python3.10/unittest/mock.py:2135: RuntimeWarning: coroutine 'WorkerHandler.process_files_from_master' was never awaited
        setattr(_type, entry, MagicProxy(entry, self))
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/cluster/tests/test_worker.py::test_worker_handler_setup_receive_files_from_master
      /usr/lib/python3.10/unittest/mock.py:2181: RuntimeWarning: coroutine 'Handler.log_exceptions' was never awaited
        self.name = name
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/tests/test_configuration.py::test_get_active_configuration[000-agent-agent-sockets-ok {"agent": {"enabled": "yes"}}]
      /usr/lib/python3.10/unittest/mock.py:2181: RuntimeWarning: coroutine 'WorkerHandler.process_files_from_master' was never awaited
        self.name = name
      Enable tracemalloc to get traceback where the object was allocated.
      See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    
    wazuh/core/tests/test_security.py::test_invalid_roles_tokens[role_list0-expected_roles0]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04aeb630a0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/core/tests/test_security.py::test_check_relationships[role_list1-expected_users1]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac497e20>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/core/tests/test_security.py::test_rbac_db_factory_reset
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac273e20>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/rbac/tests/test_auth_context.py::test_auth_roles
      /home/fdalmau/git/wazuh/framework/wazuh/rbac/auth_context.py:225: FutureWarning: Possible nested set at position 2
        regex = re.compile(regex)
    
    wazuh/rbac/tests/test_orm.py::test_add_token
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/python.py:198: PytestReturnNotNoneWarning: Expected None, but wazuh/rbac/tests/test_orm.py::test_add_token returned ([106, 107], [106, 107, 108]), which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
        warnings.warn(
    
    wazuh/rbac/tests/test_orm.py::test_add_user_roles
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/python.py:198: PytestReturnNotNoneWarning: Expected None, but wazuh/rbac/tests/test_orm.py::test_add_user_roles returned ([100, 101], [100, 101]), which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
        warnings.warn(
    
    wazuh/rbac/tests/test_orm.py::test_add_role_rule
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/python.py:198: PytestReturnNotNoneWarning: Expected None, but wazuh/rbac/tests/test_orm.py::test_add_role_rule returned ([100, 101], [100, 101]), which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
        warnings.warn(
    
    wazuh/rbac/tests/test_orm.py::test_add_role_policy
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/python.py:198: PytestReturnNotNoneWarning: Expected None, but wazuh/rbac/tests/test_orm.py::test_add_role_policy returned ([100, 101], [100, 101]), which will be an error in a future version of pytest.  Did you mean to use `assert` instead of `return`?
        warnings.warn(
    
    wazuh/tests/test_security.py::test_security[set_role_policy-params4-expected_result4]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ae9af250>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[get_users-params9-expected_result9]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ae854c10>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[create_user-params15-expected_result15]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04a7ebb760>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[remove_users-params21-expected_result21]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac59dd80>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[get_roles-params27-expected_result27]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac6af1c0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[update_role-params33-expected_result33]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac040310>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[edit_run_as-params39-expected_result39]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04a7740280>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[get_policies-params45-expected_result45]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04a7129e10>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[update_policy-params51-expected_result51]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04a7129360>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_security[remove_policies-params57-expected_result57]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ac111bd0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_rbac_catalog[get_rbac_actions-params0-expected_result0]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ae48a3b0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_security.py::test_rbac_catalog[get_rbac_resources-params5-expected_result5]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04a71c45e0>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    wazuh/tests/test_syscollector.py::test_failed_get_item_agent[agent_list0-1701]
      /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f04ae66d120>
      
      Traceback (most recent call last):
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
          lambda ref: _finalize_fairy(
        File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1031, in _finalize_fairy
          fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
      AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
      
        warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    
    -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

</details>

Some of these are not direct consequences of the update and are already contemplated in #13267.

**Comment by fdalmaup** [comment]
# Issue Update

From the previous logs, the tests involved in the `PytestUnraisableExceptionWarning` warning messages are
```
scripts/tests/test_rbac_control.py::test_restore_default_passwords[]
wazuh/core/cluster/tests/test_common.py::test_rft_init
wazuh/core/tests/test_security.py::test_invalid_roles_tokens[role_list0-expected_roles0]
wazuh/core/tests/test_security.py::test_check_relationships[role_list1-expected_users1]
wazuh/core/tests/test_security.py::test_rbac_db_factory_reset
wazuh/tests/test_security.py::test_security[set_role_policy-params4-expected_result4]
wazuh/tests/test_security.py::test_security[get_users-params9-expected_result9]
wazuh/tests/test_security.py::test_security[create_user-params15-expected_result15]
wazuh/tests/test_security.py::test_security[remove_users-params21-expected_result21]
wazuh/tests/test_security.py::test_security[get_roles-params27-expected_result27]
wazuh/tests/test_security.py::test_security[update_role-params33-expected_result33]
wazuh/tests/test_security.py::test_security[edit_run_as-params39-expected_result39]
wazuh/tests/test_security.py::test_security[get_policies-params45-expected_result45]
wazuh/tests/test_security.py::test_security[update_policy-params51-expected_result51]
wazuh/tests/test_security.py::test_security[remove_policies-params57-expected_result57]
wazuh/tests/test_security.py::test_rbac_catalog[get_rbac_actions-params0-expected_result0]
wazuh/tests/test_security.py::test_rbac_catalog[get_rbac_resources-params5-expected_result5]
wazuh/tests/test_syscollector.py::test_failed_get_item_agent[agent_list0-1701]
```
These are making use of the `rbac/orm`. Further investigation is in progress.

**Comment by fdalmaup** [comment]
# Issue Update

The warning messages were generated for not closing the sessions and engines from `orm.db_manager` used in the `db_setup` function from the `wazuh/tests/test_security.py` file which is used in several tests in order to create the RBAC database.

After modifying the function and its uses in the tests, only the following warning related to the mentioned cause remains when running every `framework` test:
```
wazuh/tests/test_security.py::test_security[set_role_policy-params0-expected_result0]
  /home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function _ConnectionRecord.checkout.<locals>.<lambda> at 0x7f23aca5e7a0>
  
  Traceback (most recent call last):
    File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 732, in <lambda>
      lambda ref: _finalize_fairy(
    File "/home/fdalmau/venv/deps-3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1043, in _finalize_fairy
      fairy.dbapi_connection = fairy._connection_record = None  # type: ignore
  AttributeError: 'NoneType' object has no attribute 'dbapi_connection'
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

```

**Comment by fdalmaup** [comment]
# Issue Update

Added teardown code after the `yield` inside `db_setup` to avoid code repetition. This didn't solve the warning message. It's relevant to mention that the message does not only correspond to `wazuh/tests/test_security.py::test_security[set_role_policy-params0-expected_result0]`, i.e., with only that set of parameters for the test. It also happens with:
```
wazuh/tests/test_security.py::test_security[set_role_policy-params4-expected_result4]
wazuh/tests/test_security.py::test_security[set_role_policy-params1-expected_result1]
wazuh/tests/test_security.py::test_security[set_role_policy-params3-expected_result3]
wazuh/tests/test_security.py::test_security[set_role_policy-params2-expected_result2]
```

**Comment by fdalmaup** [comment]
# Issue Update

Modified the code to close and dispose of the sessions and engine of `orm.db_manager` before making a new connection in `db_setup` and making a call to `orm.db_manager.close_sessions()` after clearing the mappers: 
https://github.com/wazuh/wazuh/blob/04b93fe743f54118a1d084904c575c9187abea46/framework/wazuh/tests/test_security.py#L70-L87

This generated the last warning message. No more related warning messages have been found.

The next step is to generate the Wazuh embedded Python interpreter files for `amd64` and `aarch64` after rebasing to the `feature/13265-upgrade-python-to-3.10.11` (#13265 epic branch)

**Comment by fdalmaup** [comment]
# Issue Update

Successful installations have been carried out in `amd64` and `aarch64`. No other warnings were found. 
The issue will be On Hold until the latest version of the Wazuh embedded Python interpreter has the latest updates regarding its dependencies.

**Comment by fdalmaup** [comment]
# Issue Update

The Wazuh embedded interpreter files needed to be rebuilt due to the upcoming Wazuh dependencies update (https://github.com/wazuh/wazuh/issues/16279). The installations in amd64 and aarch64 did not present any inconvenience.

**Comment by fdalmaup** [comment]
# Issue Update

On June 6, 2023, the patch version `3.10.12` was released. The CPython sources and preinstalled packages were made for `amd64` and `aarch64`, carrying out successful installations for both architectures. 

The issue goes back to Blocked until https://github.com/wazuh/wazuh-packages/issues/2154 is merged.

**Comment by fdalmaup** [comment]
# Issue Update

A rebase against the `feature/13265-upgrade-python-to-3.10.13` branch was made. Only one error arose in the Framework UTs: 
```
FAILED framework/wazuh/core/cluster/tests/test_common.py::test_handler_msg_build_ok - AssertionError: assert 10 == 13
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================================== 1 failed, 175 passed, 21 warnings in 4.27s ================================================================================
```

This was already fixed [here](https://github.com/wazuh/wazuh/issues/13267#issuecomment-1560133875).

The issue will be continued once #13267 is merged.

**Comment by fdalmaup** [comment]
# Issue Update

Rebased to branch `feature/13265-upgrade-python-to-3.10.13` and regenerated the packages of the embedded interpreter in order to have them ready to test the `fix/16679-update-sqlalchemy-2.0.0` dev branch with the latest modifications.

**Comment by fdalmaup** [comment]
# Issue Update

- Rebased to branch `feature/13265-upgrade-python-to-3.10.13` after the merge of `fix/13267-Python3.10-test-warnings`.
- Solved some new warnings in the Framework unit tests.

**Comment by fdalmaup** [comment]
# API endpoints RBAC ITs

## Test report
[API endpoints RBAC ITs.zip](https://github.com/wazuh/wazuh/files/13226792/API.endpoints.RBAC.ITs.zip)

## Test results

---
| **Test name** | **Pass** | **XPass** | **Skip** | **XFail** | **Fail** | **Issues Ref.** | **Status** |
|--|--|--|--|--|--|--|--|
| test_rbac_black_agent_endpoints.tavern.yaml | 42 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_ciscat_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_experimental_endpoints.tavern.yaml | 11 | 0 | 0 | 0 | 1 | Check note below | :red_circle: |
| test_rbac_black_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_overview_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_vulnerability_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_webhook_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_agent_endpoints.tavern.yaml | 42 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_all_endpoints.tavern.yaml | 167 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_ciscat_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_experimental_endpoints.tavern.yaml | 11 | 0 | 0 | 0 | 1 | Check note below | :red_circle: |
| test_rbac_white_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_overview_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_vulnerability_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_webhook_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |

The error in `test_rbac_black_experimental_endpoints.tavern.yaml` and `test_rbac_white_experimental_endpoints.tavern.yaml` tests is related to the problem described in #19722. It has already been fixed for branch `4.7.0` but the upward merge to `4.8.0` has not yet been carried out.

**Comment by Selutario** [comment]
GJ @fdalmaup
